plotly/tools.py

```diff
@@ -363,18 +366,22 @@ def embed(file_owner_or_url, file_id=None, width="100%", height=525):
     file_id (default=None) -- an int or string that can be converted to int
                               if you're using a url, don't fill this in!
     width (default="100%") -- an int or string corresp. to width of the figure
-    height (default="525") -- same as width but corresp. to the height of the figure
+    height (default="525") -- same as width but corresp. to the height of the
+                              figure
 
     """
     try:
-        s = get_embed(file_owner_or_url, file_id, width, height)
+        s = get_embed(file_owner_or_url, file_id=file_id, width=width,
+                      height=height)
+
         # see if we are in the SageMath Cloud
         from sage_salvus import html
         return html(s, hide=False)
     except:
         pass
     if _ipython_imported:
-        if file_id:
+        if (file_id and
+                get_config_file()['plotly_domain'] not in file_owner_or_url):
```

Comment:
The old call looked like [this](https://github.com/plotly/python-api/blob/master/plotly/plotly/plotly.py#L136):

`return tools.embed(username, plot_id, **embed_options)`

Note that we're sending `username` and `plot_id` *not* `url`. It's *better* to use these since the logic for this case is simpler, but *just* sending `url` (no `file_id`) is OK too.